### PR TITLE
Fix: updated parsing for brokerConfig and brokerSalesInfo

### PR DESCRIPTION
**Pull request by piggydoughnut**

### CODE REVIEW

File: packages/react-hooks/src/useBrokerConfig.ts [hunk 1]
```diff
@@ -11,17 +11,17 @@ import { createNamedHook, useApi, useCall } from '@polkadot/react-hooks';
 import { stringToBN } from './utils/dataProcessing.js';
 
 function parseConfig (config: PalletBrokerConfigRecord): SimplifiedPalletBrokerConfigRecord {
-  const c = config.toJSON();
+  const c = config?.toJSON();
```
Comment (TarikGul) [inline packages/react-hooks/src/useBrokerConfig.ts:14]: This should actually be like this:

```typescript
function parseConfig (config: PalletBrokerConfigRecord): SimplifiedPalletBrokerConfigRecord {
  return {
    advanceNotice: config.advanceNotice.toNumber(),
    contributionTimeout: config.contributionTimeout.toNumber(),
    idealBulkProportion: stringToBN(config.idealBulkProportion?.toString()),
    interludeLength: config.interludeLength.toNumber(),
    leadinLength: config.leadinLength.toNumber(),
    limitCoresOffered: config.limitCoresOffered.isSome ? config.limitCoresOffered.unwrap().toNumber() : 0,
    regionLength: config.regionLength.toNumber(),
    renewalBump: stringToBN(config.renewalBump?.toString())
  };
}
```

Comment (TarikGul) [reply packages/react-hooks/src/useBrokerConfig.ts:14]: And if config can be undefined, then the param type needs to be updated.

File: packages/react-hooks/src/useBrokerSalesInfo.ts [hunk 2]
```diff
@@ -10,22 +10,20 @@ import { createNamedHook, useApi, useCall } from '@polkadot/react-hooks';
 
 import { stringToBN } from './utils/dataProcessing.js';
 
-function parsePalletBrokerSaleInfoRecord (
-  record: PalletBrokerSaleInfoRecord
-): SimplifiedPalletBrokerSaleInfoRecord {
-  const rec = record.toJSON();
+function parsePalletBrokerSaleInfoRecord (record: PalletBrokerSaleInfoRecord): SimplifiedPalletBrokerSaleInfoRecord {
```
Comment (TarikGul) [inline packages/react-hooks/src/useBrokerSalesInfo.ts:13]: Same with this as above.

**Review by TarikGul — CHANGES_REQUESTED** [review]
It's always better to use the built in type structures, instead of calling `toJSON` in order to get all the values.

File: packages/react-hooks/src/useBrokerSalesInfo.ts [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    coresOffered: record.coresOffered.toNumber(),
+    coresSold: record.coresSold.toNumber(),
+    endPrice: record.endPrice,
+    firstCore: record.firstCore.toNumber(),
+    idealCoresSold: record.idealCoresSold.toNumber(),
+    leadinLength: record.leadinLength.toNumber(),
+    regionBegin: record.regionBegin.toNumber(),
+    regionEnd: record.regionEnd.toNumber(),
+    saleStart: record.saleStart.toNumber(),
+    selloutPrice: record.selloutPrice.isSome ? record.selloutPrice.unwrap() : 0
```
Comment (TarikGul) [inline packages/react-hooks/src/useBrokerSalesInfo.ts:22]: Small nit, but I think you can set the `0` to `new BN(0)` if you want the values to be inline. That being said its a small nit.

**Review by TarikGul — APPROVED** [review]

File: packages/react-hooks/src/useBrokerSalesInfo.ts (hunk 3 above)
Comment (piggydoughnut) [reply packages/react-hooks/src/useBrokerSalesInfo.ts]: good nit 🤩

**Pull request merged**